hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1484,6 +1515,125 @@ def single_greedy_shrink_iteration(self):
 
         self.lower_common_block_offset()
 
+    # A block program describes a transformation on a short sequence of blocks
+    # that we can run over all of the blocks, with the hope of it shrinking the
+    # target. See run_block_program below for more details.
+    #
+    # We have two sets of standard block programs. These have mostly been uh
+    # "empircally determined" (I generated all short block programs and ran a
+    # bunch of them on a thorny test case to see which ones worked well). We
+    # also have some that we know *might* be useful but we don't expect usually
+    # are.
+
+    HELPFUL_BLOCK_PROGRAMS = [
+        '-X-',
+        '-XX',
+        '0-X',
+        '-XX',
+        '00',
+    ]
+
+    LESS_HELPFUL_BLOCK_PROGRAMS = [
+        'XXXX',
+        'XXX',
+        'XX',
+        '-XX',
```

Comment:
This one looks as helpful and less helpful at the same time. That's not a typo?

Reply:
Oh, and it's in helpful list twice.

Reply:
Ugh, thanks.